server/slack_notifier.go

```diff
@@ -0,0 +1,147 @@
+	"fmt"
+
+	"github.com/applikatoni/applikatoni/deploy"
+	"github.com/applikatoni/applikatoni/models"
+
+	"log"
+	"net/http"
+	"text/template"
+)
+
+const slackSuccessSummaryTmplStr = `{{.GitHubRepo}} Successfully Deployed:
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+const slackFailSummaryTmplStr = `:rotating_light:  {{.GitHubRepo}} Deployed Failed :rotating_light: :
+{{.Username}} deployed {{.Branch}} on {{.Target}} :pizza:
+
+> {{.Comment}}
+<{{.GitHubUrl}}|View latest commit on GitHub>
+<{{.DeploymentURL}}|Open deployment in Applikatoni>`
+
+var slackSuccessSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackSuccessSummaryTmplStr))
+var slackFailSummaryTemplate = template.Must(template.New("slackSuccessSummary").Parse(slackFailSummaryTmplStr))
+
+type slackMsg struct {
+	Text string `json:"text"`
+}
+
+func NotifySlack(db *sql.DB, deploymentId int, success bool) {
+	deployment, err := getDeployment(db, deploymentId)
+	if err != nil {
+		log.Printf("Could not find deployment with id %v, %s\n", deploymentId, err)
+		return
+	}
+
+	application, err := findApplication(deployment.ApplicationName)
+	if err != nil {
+		log.Printf("Could not find application with name %v, %s\n", deployment.ApplicationName, err)
+		return
+	}
+
+	target, err := findTarget(application, deployment.TargetName)
+	if err != nil {
+		log.Printf("Could not find target with name %v, %s\n", deployment.TargetName, err)
+		return
+	}
+
+	user, err := getUser(db, deployment.UserId)
+	if err != nil {
+		log.Printf("Could not find User with id %v, %s\n", deployment.UserId, err)
+		return
+	}
+
+	if target.SlackUrl == "" {
+		return
+	}
+
+	var summary string
+	if success {
+		summary, _ = generateSlackSummary(deployment, application, user, slackSuccessSummaryTemplate)
+	} else {
+		summary, _ = generateSlackSummary(deployment, application, user, slackFailSummaryTemplate)
+	}
+	if err != nil {
+		log.Printf("Could not generate Slack deployment summary, %s\n", err)
+		return
+	}
+
+	SendSlackRequest(deployment, target, application, summary)
+}
+
+func generateSlackSummary(d *models.Deployment, a *models.Application, u *models.User, msgTmpl *template.Template) (string, error) {
+	var summary bytes.Buffer
+
+	var protocol string
+	if config.SSLEnabled {
+		protocol = "https"
+	} else {
+		protocol = "http"
+	}
+
+	deploymentUrl := fmt.Sprintf("%s://%s/%v/deployments/%v",
+		protocol,
+		config.Host,
+		a.GitHubRepo, d.Id)
+
+	gitHubUrl := fmt.Sprintf("https://github.com/%v/%v/commit/%v",
+		a.GitHubOwner, a.GitHubRepo, d.CommitSha)
+
+	err := msgTmpl.Execute(&summary, map[string]interface{}{
+		"GitHubRepo":    a.GitHubRepo,
+		"Branch":        d.Branch,
+		"Target":        d.TargetName,
+		"Username":      u.Name,
+		"Comment":       d.Comment,
+		"GitHubUrl":     gitHubUrl,
+		"DeploymentURL": deploymentUrl,
+	})
```

Comment:
I would like one generateSummary function for slack and flowdock.
So the struct should be called reportMsg or so and be moved out of flowdock and slack.
Maybe some new file report.go with this struct and the generate function in it.
I will do that change and a pull request later if you like.